packages/legend-application-studio/src/stores/editor/editor-state/element-editor-state/ClassState.ts

```diff
@@ -104,7 +104,10 @@ export class DerivedPropertyState extends LambdaEditorState {
     }
   }
 
-  *convertLambdaObjectToGrammarString(pretty: boolean): GeneratorFn<void> {
+  *convertLambdaObjectToGrammarString(
+    pretty: boolean,
+    clearErrorSourceInformation?: boolean | undefined,
```

Comment:
maybe the flag name should be more specific to error. I think the point of the flag is we want to keep the error. 
so maybe 
`keepError?: boolean`
then there we keep error but clear source info